src/com/lftechnology/java/training/krishna/employeemanagement/dao/user/impl/UserDaoImpl.java

```diff
@@ -0,0 +1,104 @@
+
+package com.lftechnology.java.training.krishna.employeemanagement.dao.user.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.lftechnology.java.training.krishna.employeemanagement.dao.user.UserDao;
+import com.lftechnology.java.training.krishna.employeemanagement.domain.User;
+import com.lftechnology.java.training.krishna.employeemanagement.jdbc.DbConnectionFactory;
+import com.lftechnology.java.training.krishna.employeemanagement.utils.ConstantUtils;
+
+/**
+ * Implementation of user DAO interface.
+ *
+ * @author Krishna Timilsina <krishnatimilsina@lftechnology.com>
+ * @version 1.0
+ */
+public class UserDaoImpl implements UserDao {
+
+	private static final Logger LOGGER =
+		Logger.getLogger(UserDaoImpl.class.getName());
+
+	Connection connection = DbConnectionFactory.getDbConnection();
+	PreparedStatement preparedStatement = null;
+	ResultSet rs = null;
+	private User activeUser;
+
+	private static final String GET_USER_QUERY =
+		"SELECT * FROM user WHERE username = ? AND password = ? AND isTerminated = ?";
+
+	public UserDaoImpl() {
+
+	}
+
+	@Override
+	public User authenticate(User entity) {
+
+		activeUser = null;
+		try {
+			preparedStatement = connection.prepareStatement(GET_USER_QUERY);
+			preparedStatement.setString(1, entity.getUsername());
+			preparedStatement.setString(2, entity.getPassword());
+			preparedStatement.setInt(3, ConstantUtils.ACTIVE);
+			ResultSet resultSet = preparedStatement.executeQuery();
+
+			if (resultSet.next()) {
+				activeUser =
+					new User(
+						resultSet.getInt("id"),
+						resultSet.getString("username"),
+						resultSet.getString("password"),
+						resultSet.getInt("isTerminated"),
+						resultSet.getDate("createdAt"),
+						resultSet.getDate("modifiedAt"));
+			}
+		}
+		catch (SQLException e) {
+			LOGGER.log(
+				Level.SEVERE, "Exception while creating statement: {0}", e);
+		}
+		finally {
+			DbConnectionFactory.closePreparedStatement();
+		}
+		return activeUser;
+	}
+
+	@Override
+	public User create(User entity) {
+
+		return null;
+	}
+
+	@Override
+	public User update(User entity) {
+
+		return null;
+	}
+
+	@Override
+	public User findById(Integer id) {
+
+		return null;
+	}
+
+	@Override
+	public List<User> findAll(String key) {
+
+		List<User> user = new ArrayList<User>();
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Immediately return this expression instead of assigning it to the temporary variable "user". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1488)